web_notify/static/src/js/web_client.js

```diff
@@ -61,16 +60,26 @@ odoo.define('web_notify.WebClient', function (require) {
             });
         },
         on_message: function (message) {
-            return this.call(
-                'notification', 'notify', {
-                    type: message.type,
-                    title: message.title,
-                    message: message.message,
-                    sticky: message.sticky,
-                    className: message.className,
-                }
-            );
+        	var notif_method = 'notify';
+        	var service_class =  'notification' ;
+
+        	if(message.allow_native_notifications){
```

Comment:
I miss a call to https://developer.mozilla.org/en-US/docs/Web/API/Notification/permission here